### PR TITLE
feat(scripts): npm run verify:unusedStrapiEntries

### PR DESCRIPTION
**Pull request by chohner**
https://app.asana.com/0/1203259475859667/1208196006606406/f

- scans all xstate configs to recursively build URLs
- parses URLs from all form-flow-, vorabcheck-, result-pages (from content file)
- compares and returns all URLs that aren't referenced in xstate configs
- additionally prints pages without stepId

### CODE REVIEW

File: scripts/unusedStrapiEntries.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+  const pages = [
+    ...content["form-flow-pages"],
+    ...content["vorab-check-pages"],
+    ...content["result-pages"],
+  ];
+
+  const [pagesWithStepId, pagesWithoutStepId] = partitionPagesByStepId(pages);
+
+  if (pagesWithoutStepId.length > 0) {
```
Comment (rbrtrfl) [inline scripts/unusedStrapiEntries.ts:71]: question: should we also look for pages without `flowId`?

File: scripts/unusedStrapiEntries.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+type MinimalPage = {
+  attributes: {
+    flow_ids: StrapiSchemas["form-flow-pages"][0]["attributes"]["flow_ids"];
+    stepId: StrapiSchemas["form-flow-pages"][0]["attributes"]["stepId"];
+  };
+};
+
+// recursivly traverse states object, while concatenating nested names. Returns a flat list
+function allStateNames(xstateConfigStates: Config["states"]): string[] {
```
Comment (rbrtrfl) [inline scripts/unusedStrapiEntries.ts:18]:
```suggestion
function getAllStateNames(xstateConfigStates: Config["states"]): string[] {
```

File: scripts/unusedStrapiEntries.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        );
+      } else {
+        // stepId in xstate config use ergebnis/ prefix to indicate ErgebnisPage
+        return stateName.replace("ergebnis/", "");
+      }
+    },
+  );
+}
+
+function urlsFromPages(pages: MinimalPage[]) {
```
Comment (rbrtrfl) [inline scripts/unusedStrapiEntries.ts:35]:
```suggestion
function extractUrlsFromPages(pages: MinimalPage[]) {
```

File: scripts/unusedStrapiEntries.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      (flowId) => flowId.attributes.flowId + page.attributes.stepId,
+    ),
+  );
+}
+
+function partitionPagesByStepId(pages: MinimalPage[]) {
+  return _.partition(pages, (page) => page.attributes.stepId !== null);
+}
+
+async function unusedStrapiEntry() {
```
Comment (rbrtrfl) [inline scripts/unusedStrapiEntries.ts:48]:
```suggestion
async function findUnusedStrapiEntries() {
```

**Review by rbrtrfl — APPROVED** [review]

File: scripts/unusedStrapiEntries.ts (hunk 2 above)
Comment (chohner) [reply scripts/unusedStrapiEntries.ts:18]: I know we don't really have a naming convention but I try to use nouns when returning said thing and verbs if state is changed.

**Pull request merged**